wiki/development/coding-guidelines.md

```diff
@@ -80,127 +238,233 @@ class Two {foo = 2;};
 
-#### 4.2.2 STRING family macros
+Good:
+```
+_myVariable = 0;
+{
+    _x params ["_value", "_amount"];
+    if (_value > 0) then {
+        _myVariable = _myVariable + _amount;
+    };
+} forEach _array;
+```
+Bad:
+```
+_myvariable = 0;
+if (_condition) then {
+    _myVariable = 1;
+} else {
+    _myvariable = 2;
+};
+```
+Good:
+```
+_myvariable = [1, 2] select _condition;
+```
 
-Note that you need the strings in module stringtable.xml in the correct format
-`STR_ACE_<module>_<string>`</br>
-Example:</br>
-`STR_Balls_Banana`</br>
+### 6.9. Initialization expression in for loops
+The initialization expression in a for loop will perform no actions other than to initialize the value of a single for loop parameter.
 
-Script strings:
+### 6.10. Increment expression in for loops
+The increment expression in a for loop will perform no action other than to change a single loop parameter to the next value for the loop.
 
-| Macro  |  Expands to |
-| -------|---------|
-| `LSTRING(banana)` |  "STR_ACE_balls_banana"|
-| `ELSTRING(balls,banana)` | "STR_ACE_balls_banana"|
+### 6.11. GetVariable
+When using getvariable, there should either be a default value  given in the statement or the return value should be checked for correct data type as well as return value. A default value may not be given after a nil check.
+
+Bad:
+```
+_return = obj getvariable "varName";
+if (!isnil "_return") then {_return = 0 };
+```
+Good:
+```
+_return = obj getvariable ["varName", 0];
+```
+Good:
+```
+_return = obj getvariable "varName";
+if (!isnil "_return") exitwith {};
+```
 
+### 6.12. Global Variables
+Global variables should not be used to pass along information from one function to another. Use arguments instead.
 
-Config Strings (require `$` as first character):
+### 6.13. Temporary Objects & Variables
+Unnecessary temporary objects or variables should be avoided.
 
-| Macro  |  Expands to |
-| -------|---------|
-| `CSTRING(banana)` |  "$STR_ACE_balls_banana" |
-| `ECSTRING(balls,banana)` | "$STR_ACE_balls_banana" |
+### 6.14. Commented out Code
+Code that is not used (commented out) shall be deleted.
+
+### 6.15. Constant Global Variables
+All global variables that are intended to be used as a constant shall be written in all caps with underscores between the words, with the exception of the prefix. This should be done through the usage of the `GVAR` macro family.
+
+Good:
+```
+ACE_Common_CATEGORY_NAME = 1;
+GVAR(CATEGORY_NAME) = 1
+```
+
+### 6.16. Logging
+Functions should whenever possible and logical, make use of logging functionality through the logging and debugging macros from CBA and ACE.
+
+### 6.17. Constant Private Variables
+Constant private variables that are used more as once should be put in a #define.
 
+### 6.18. Code used more than once
+Any piece of code that could/is used more than once, should be put in a function and it’s seperate .sqf file, unless this code is less as 5 lines and used only in a per frame handler.
```

Comment:
Any piece of code that could/is used more than once, should be put in a function and its separate `.sqf` file, unless this code is less than 5 lines and used only in a per-frame handler.